algorithm-simulator/src/main/scala/style95/generator/SingleBehaviorActor.scala

```diff
@@ -0,0 +1,81 @@
+package style95.generator
+
+import akka.actor.{Actor, ActorLogging, ActorRef, PoisonPill, Props, Timers}
+import style95.Container.ActivationMessage
+
+import scala.collection.immutable.Queue
+import scala.concurrent.duration._
+import scala.language.postfixOps
+
+object SingleBehaviorActor {
+  def props(behavior: IntervalBehavior, receiver: ActorRef): Props =
+    Props(new SingleBehaviorActor(behavior, receiver))
+
+  final case object SendRequest
+  private case object TickKey
+  private case object GenerateHits
+}
+
+class SingleBehaviorActor(behavior: IntervalBehavior, receiver: ActorRef)
+    extends Actor
+    with ActorLogging
+    with Timers {
+  import SingleBehaviorActor._
+  import context.{dispatcher, system}
+
+  private var timingGen: TimingGenerator = _
+  private var start: Long = _
+  private var preGeneratedHits: Queue[Long] = _
+
+  // Generate hits at frequently as possible. The more frequent ticks are generated,
+  // the closer they conform to Poisson distribution
+  private val schedulerPrecision =
+    context.system.settings.config.getDuration("akka.scheduler.tick-duration")
+  log.info(s"the timing accuracy of the actor scheduler is $schedulerPrecision")
+  timers.startPeriodicTimer(TickKey, GenerateHits, schedulerPrecision)
+
+  override def preStart(): Unit = {
+    // Reset state on start/restart
+    timingGen = behavior.behavior.timing
```

Comment:
Is there any case this actor is restarted?
If not, it would be better to use class parameters or initialize fields in the constructor?

Reply:
The `preStart` may be used later.  In the future we might want simulate a long running scenario with some repetitive sequences. One method is to use a loop to build many actors and start them one by one. Another method is to restart the actors; thus, reset their internal states.

Reply:
Then I think it would be safe to take the functional way here.
One option is we can safely rely on immutable variables with creating the same actors for repetition instead of manipulating mutable variables directly.
Or we can just terminate actors with supervision.
Actors will be restarted based on supervision, but I prefer the first option.

Reply:
Indeed, creating a new actor is a safer choice than reusing an existing actor. I will move the initialization logic to the class constructor.